agent/recorder-agent.c

```diff
@@ -0,0 +1,385 @@
+/**
+ *  Copyright 2019 SK Telecom Co., Ltd.
+ *    Author: Walter Lozano <walter.lozano@collabora.com>
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "config.h"
+
+#include "recorder-agent.h"
+#include <hwangsae/recorder.h>
+#include <gaeguli/gaeguli.h>
+#include <gio/gio.h>
+#include <glib/gstdio.h>
+#include <gst/pbutils/pbutils.h>
+#include <curl/curl.h>
+
+#include <glib-unix.h>
+
+#include <hwangsae/dbus/manager-generated.h>
+#include <hwangsae/dbus/recorder-interface-generated.h>
+
+#define RECORDER_GAEGULI_SRT_MODE GAEGULI_SRT_MODE_CALLER
+
+#define HWANGSAE_RECORDER_SCHEMA_ID "org.hwangsaeul.hwangsae.recorder"
+
+struct _HwangsaeRecorderAgent
+{
+  GApplication parent;
+
+  HwangsaeRecorder *recorder;
+  Hwangsae1DBusManager *manager;
+  Hwangsae1DBusRecorderInterface *recorder_interface;
+  GSettings *settings;
+
+  gboolean is_recording;
+  gboolean got_file_created_signal;
+  gboolean got_file_completed_signal;
+};
+
+/* *INDENT-OFF* */
+G_DEFINE_TYPE (HwangsaeRecorderAgent, hwangsae_recorder_agent, G_TYPE_APPLICATION)
+/* *INDENT-ON* */
+
+typedef enum
+{
+  RELAY_METHOD_NONE = 0,
+  RELAY_METHOD_START_STREAMING,
+  RELAY_METHOD_STOP_STREAMING
+} relay_methods;
+
+static void
+hwangsae_recorder_agent_send_rest_api (HwangsaeRecorderAgent * self,
+    relay_methods method, gchar * edge_id)
+{
+  CURL *curl;
+  g_autofree gchar *host =
+      g_settings_get_string (self->settings, "relay-address");
+  guint port = g_settings_get_uint (self->settings, "relay-api-port");
```

Comment:
You might want to use `g_settings_bind()` to sync with gsettings, but not for this commit.